### PR TITLE
fix validation issues; fix typo; rm Add Asset btn

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: extension/src/popup/components/manageAssets/ChooseAsset/index.tsx [hunk 1]
```diff
@@ -120,13 +120,15 @@ export const ChooseAsset = ({
...
-        </div>
+        {!selectingAssetType && (
+          <div className="ChooseAsset__button">
+            <Link to={ROUTES.searchAsset}>
+              <Button fullWidth variant={Button.variant.tertiary}>
+                {t("Add another asset")}
+              </Button>
+            </Link>
+          </div>
+        )}
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ChooseAsset/index.tsx:131]: Remove the `Add Another Asset` btn if we're choosing an asset in a path payment

File: extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx [hunk 2]
```diff
@@ -138,6 +143,14 @@ export const ManageAssetRows = ({
...
         );
+        trackChangeTrustline();
+        dispatch(resetSubmission());
+        navigateTo(ROUTES.account);
+      }
+
+      if (submitFreighterTransaction.rejected.match(submitResp)) {
+        setErrorAsset(assetSubmitting);
+        navigateTo(ROUTES.trustlineError);
       }
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx:154]: Handling a trustline error fell off in this release. Restoring it and refactor a bit to track properly

File: extension/src/popup/components/sendPayment/SendAmount/index.tsx [hunk 3]
```diff
@@ -416,7 +416,10 @@ export const SendAmount = ({
             <div className="SendAmount__btn-continue">
               <Button
                 disabled={
-                  loadingRate || formik.values.amount === "0" || !formik.isValid
+                  loadingRate ||
+                  formik.values.amount === "0" ||
+                  !formik.isValid ||
+                  !destinationAmount
```
Comment (piyalbasu) [inline extension/src/popup/components/sendPayment/SendAmount/index.tsx:422]: Just noticed this bug: in a path payment, if there's no path available (for ex: using SRT in testnet), it still allows you to continue. Let's disable that

File: extension/src/popup/views/RecoverAccount/index.tsx [hunk 4]
```diff
@@ -218,7 +222,13 @@ export const RecoverAccount = () => {
                     <Button
                       fullWidth
                       isLoading={isSubmitting}
-                      disabled={!(dirty && isValid)}
+                      disabled={
+                        !(
+                          dirty &&
+                          isValid &&
+                          buildMnemonicPhrase(mnemonicPhraseArr).length
```
Comment (piyalbasu) [inline extension/src/popup/views/RecoverAccount/index.tsx:229]: When Recovering using a mnemonic phrase, we need to manually check that the user has entered a phrase bc these are custom fields now, not Formik

**Review by acharb — APPROVED** [review]
lgtm!

**Pull request merged**